agent/agent_command_test.go

```diff
@@ -154,6 +154,7 @@ func (s *CommandTestSuite) TestTimeout() {
 			ID:     taskID,
 			Secret: taskSecret,
 		},
+		timedOut: true,
```

Comment:
I don't remember the details of this test, but shouldn't this bool get set as a result of running the test?

Reply:
I think we're testing more that "when we timeout the right things happen" not "if we get ourselves into a situation where we timeout, that we timeout and then the right things happen," though if I missed that intention, then we should write more tests and split this up, potentially. 

I may also be totally wrong.